0.68 changelog (#24355)
microsoft/PowerToys: README.md

Our [prioritized roadmap][roadmap] of features and utilities that the core team is focusing on.

### 0.68 - February 2023 Update

In this release, we focused on releasing new features, stability and improvements.

**Highlights**

- New utility: Paste as Plain Text allows pasting the text contents of your clipboard without formatting. Note: the formatted text in the clipboard is replaced with the unformatted text. Thanks [@carlos-zamora](https://github.com/carlos-zamora)!

> - New utility: Mouse Jump allows to quickly move the mouse pointer long distances on single or across screens. Thanks [@mikeclayton](https://github.com/mikeclayton)!

Reviewer: ```suggestion
- New utility: Mouse Jump allows to quickly move the mouse pointer long distances on a single screen or across multiple screens. Thanks [@mikeclayton](https://github.com/mikeclayton)!
```

- Add new GPO policies for automatic update downloads, update toast notifications and automatic background update check. Thanks [@htcfreek](https://github.com/htcfreek)!
- Support MSC and CPL files in "Run command" results of PowerToys Run Program plugin. Thanks [@htcfreek](https://github.com/htcfreek)!

> - Add support for log2 and log10 in PowerToys Run Calculator plugin. Thanks [@RickLuiken](https://github.com/RickLuiken)!

Reviewer: > There are current page which says

Grammar (single or plural?)

### General

- Improve metered network detection in runner. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
- Update PowerToys logo used by installer. Thanks [@ChaseKnowlden](https://github.com/ChaseKnowlden1)!
- Add new GPO policies for automatic update downloads, update toast notifications and automatic background update check. Thanks [@htcfreek](https://github.com/htcfreek)!
- Update copyright year to 2023. Thanks [@ChaseKnowlden](https://github.com/ChaseKnowlden)!

### FancyZones

- Refactored and improved code quality.
- Fix crashing on moving window between monitors with Win + arrows. (This was a hotfix for 0.67)
- Fix issue causing window attributes to not be reset properly. (This was a hotfix for 0.67)
- Fix issue causing window to not be adjusted when layout is changed. (This was a hotfix for 0.67)
- Fix issue causing window not to be unsnapped on drag started. (This was a hotfix for 0.67)
- Fix issue causing layouts not to be applied to new virtual desktops. (This was a hotfix for 0.67)

### File explorer add-ons

- Add Developer files previewer option to set max file size and fix styling issue. [@Aaron-Junker](https://github.com/Aaron-Junker)!

> - Improve Developer files previewer exception handling and error messages display.

Reviewer: ```suggestion
- Improve Developer files previewer exception handling and printing of error messages.
```

- Fix crash when generating PDF and Gcode files thumbnails. (This was a hotfix for 0.67)

### Hosts file editor

- Improve hosts file loading. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
- Improved duplicate hosts finding. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

### Keyboard Manager

- Fix typo in Keyboard Manager Editor. Thanks [@ChristianLW](https://github.com/ChristianLW)!

### Mouse Utils

- Resolve grammatical error in Mouse Highlighter description. Thanks [@WordlessSafe1](https://github.com/WordlessSafe1)!
- New utility: Mouse Jump allows to quickly move the mouse pointer long distances on single or across screens. Thanks [@mikeclayton](https://github.com/mikeclayton)!

### Paste as Plain Text

> - New utility: Paste as Plain Text allows pasting the text contents of your clipboard without formatting. Note: the formatted text in the clipboard is replaced with the unformatted text. Thanks [@carlos-zamora](https://github.com/carlos-zamora)!

Reviewer 2: I would put this on top

Author: what do you mean? It is listeďin highlights

Reviewer 2: Fair point

### PowerToys Run

- Show Steam (steam://open/) shortcuts in the Program plugin. 
- Localize paths of Program plugin results. Thanks [@htcfreek](https://github.com/htcfreek)!
- Support MSC and CPL files in "Run command" results of Program plugin. Thanks [@htcfreek](https://github.com/htcfreek)!

> - System plugin: Setting for separate "Open/Empty Recycle bin" results or single one with context menu. Thanks [@htcfreek](https://github.com/htcfreek)!

Reviewer: ```suggestion
- System plugin: Setting for separate "Open/Empty Recycle bin" results or single result with context menu. Thanks [@htcfreek](https://github.com/htcfreek)!
```

- Add support for log2 and log10 in Calculator plugin. Thanks [@RickLuiken](https://github.com/RickLuiken)!

> - Removed the TimeZone plugin.

Reviewer: No explanation? Or link to issue #24306? Will it come back later?

> - Improved stability of localization code. Thanks [@htcfreek](https://github.com/htcfreek)!

Reviewer 3: ```suggestion
- Improved stability of the code used to get the localized names and paths. Thanks [@htcfreek](https://github.com/htcfreek)!
```

- Fix the crash when loading thumbnail for PDF files. (This was a hotfix for 0.67)

### Shortcut Guide

> - Dismiss Shortcut Guide with mouse click. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

Reviewer: ```suggestion
- Added: Dismiss Shortcut Guide with mouse click. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
```

> - Dissmiss Shortcut Guide with mouse click. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

Author: ```suggestion
- Dismiss Shortcut Guide with mouse click. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
```

### Quick Accent

- Added Lithuanian characters. Thanks [@saulens22](https://github.com/saulens22)!
- Added additional (Chinese) characters. Thanks [@ChaseKnowlden](https://github.com/ChaseKnowlden)!

### Settings

- Add missing flyout borders on Windows 10. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

> - Add Experimentation settings landing page. Thanks [@chenss3](https://github.com/chenss3)!

Reviewer 4: i added some more detail above in a suggestion

- Add new GPO policies for automatic update downloads and update toast notifications. Thanks [@htcfreek](https://github.com/htcfreek)!
- Update copyright year to 2023. Thanks [@ChaseKnowlden](https://github.com/ChaseKnowlden)!

### FancyZones

- Refactored and improved code quality.
- Fix crashing on moving window between monitors with Win + arrows. (This was a hotfix for 0.67)
- Fix issue causing window attributes to not be reset properly. (This was a hotfix for 0.67)
- Fix issue causing window to not be adjusted when layout is changed. (This was a hotfix for 0.67)
- Fix issue causing window not to be unsnapped on drag started. (This was a hotfix for 0.67)
- Fix issue causing layouts not to be applied to new virtual desktops. (This was a hotfix for 0.67)
- Fix issues causing windows not to be restored correctly to their last known zone.

### File explorer add-ons

- Add Developer files previewer option to set max file size and fix styling issue. Thanks [@Aaron-Junker](https://github.com/Aaron-Junker)!
- Improve Developer files previewer exception handling and printing of error messages.

> - Fix crash when generating PDF and Gcode files thumbnails. (This was a hotfix for 0.67)

Reviewer 3: ```suggestion
- Fix crash when generating PDF and Gcode file thumbnails. (This was a hotfix for 0.67)
```

Don't we need "file" in this case as singular because we have thumbnails as plural? 🤔

Reviewer: "pdf and gcode" implies plural too. Better would be: "...thumbnails for pdf and gcode files"

### Hosts file editor

- Improve hosts file loading. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!
- Improved duplicate hosts finding. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

### Keyboard Manager

- Fix typo in Keyboard Manager Editor. Thanks [@ChristianLW](https://github.com/ChristianLW)!

### Mouse Utils

- Resolve grammatical error in Mouse Highlighter description. Thanks [@WordlessSafe1](https://github.com/WordlessSafe1)!
- New utility: Mouse Jump allows to quickly move the mouse pointer long distances on single or across screens. Thanks [@mikeclayton](https://github.com/mikeclayton)!

### Paste as Plain Text

- New utility: Paste as Plain Text allows pasting the text contents of your clipboard without formatting. Note: the formatted text in the clipboard is replaced with the unformatted text. Thanks [@carlos-zamora](https://github.com/carlos-zamora)!

### PowerToys Run

- Show Steam (steam://open/) shortcuts in the Program plugin. 

> - Localize paths of Program plugin results. Thanks [@htcfreek](https://github.com/htcfreek)!

Reviewer 3: ```suggestion
- Localize paths of Program plugin results. Thanks [@htcfreek](https://github.com/htcfreek)!
- Improved stability of localization code. Thanks [@htcfreek](https://github.com/htcfreek)!
```

Reviewer 4: improved stability of the localized code?

Reviewer 3: What I mean I we improved the code to get the localized strings:
- Switched to IShellItem interface.
- Implemented a cache for already localized strings to have a better performance.

This hopefully fixes the problem that it didn't work always in the past.

- Support MSC and CPL files in "Run command" results of Program plugin. Thanks [@htcfreek](https://github.com/htcfreek)!
- Added missing MSC and CPL settings to the results of Windows Settings plugin. Thanks [@htcfreek](https://github.com/htcfreek)!

> - System plugin: Setting for separate "Open/Empty Recycle bin" results or single result with context menu. Thanks [@htcfreek](https://github.com/htcfreek)!

Reviewer 3: ```suggestion
- System plugin: Setting for separate "Open/Empty Recycle bin" results or single result with context menu. (This was implemented based on user feedback for a change in the last build.) Thanks [@htcfreek](https://github.com/htcfreek)!
```

- Add support for log2 and log10 in Calculator plugin. Thanks [@RickLuiken](https://github.com/RickLuiken)!

> - Removed the TimeZone plugin.

Reviewer 3: ```suggestion
- Removed the TimeZone plugin. (We hope to bring it back again after fixing the bugs.)
```

Reviewer 4: it is more than a few bugs to bring it back.

- Fix the crash when loading thumbnail for PDF files. (This was a hotfix for 0.67)

### Shortcut Guide

- Added: Dismiss Shortcut Guide with mouse click. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

### Quick Accent

- Added Lithuanian characters. Thanks [@saulens22](https://github.com/saulens22)!
- Added additional (Chinese) characters. Thanks [@ChaseKnowlden](https://github.com/ChaseKnowlden)!

### Settings

- Add missing flyout borders on Windows 10. Thanks [@davidegiacometti](https://github.com/davidegiacometti)!

> - Add experimentation settings landing page. Thanks [@chenss3](https://github.com/chenss3)!

Reviewer 3: ```suggestion
- Add experimentation for oobe landing page. Thanks [@chenss3](https://github.com/chenss3)!
```

- Show icons of user-installed PowerToys Run plugins. Thanks [@al2me6](https://github.com/al2me6)!
- Fixed crash when clicking Browse for backup and restore location while running elevated.
- Respect taskbar position when showing system tray flyout. (This was a hotfix for 0.67)
- Show correct Hosts module image. (This was a hotfix for 0.67)

### Documentation

> - Improve GPO docs and the layout of the disabled Experiments setting. Thanks [@htcfreek](https://github.com/htcfreek)!

Reviewer 3: This was in internal fix after merging pr into dev. The wrong ui and docs was never released. Should we really mention this?

Author: good catch. Removing
